crates/circuit/src/circuit_instruction.rs

```diff
@@ -618,6 +622,51 @@ impl<'a, 'py> FromPyObject<'a, 'py> for OperationFromPython {
                     });
                 };
             }
+        } else if ob_name == "PauliProductMeasurement" {
+            let params = extract_params_no_coerce()?;
+
+            let z = if let Param::Obj(z) = &params[0] {
+                z.extract::<PyReadonlyArray1<bool>>(py)?
+                    .as_slice()?
+                    .to_vec()
+            } else {
+                return Err(PyTypeError::new_err(format!(
+                    "invalid input: {}",
+                    ob.to_owned()
+                )));
+            };
+
+            let x = if let Param::Obj(x) = &params[1] {
+                x.extract::<PyReadonlyArray1<bool>>(py)?
+                    .as_slice()?
+                    .to_vec()
+            } else {
+                return Err(PyTypeError::new_err(format!(
+                    "invalid input: {}",
+                    ob.to_owned()
+                )));
+            };
+
+            let phase = if let Param::Obj(phase) = &params[2] {
+                phase.extract::<u8>(py)?
+            } else {
+                return Err(PyTypeError::new_err(format!(
+                    "invalid input: {}",
+                    ob.to_owned()
+                )));
+            };
```

Comment:
I would say instead of params we should leave these as attributes of the Python class and not jam them in params. On unitary gate the params is a historical artifact that would be painful to change, but on a new instruction class we don't have to populate parameters and I think it would be better to use `getattr()` here to access named attributes. This has two advantages, first it doesn't imply you can easily change the pauli and not change the object but more importantly you don't have to explain why changing the parameters directly in place doesn't reflect in all cases.